From b7b8056728abdae962b55d67b136893ec2317e51 Mon Sep 17 00:00:00 2001
From: q234rty <q23456yuiop@gmail.com>
Date: Tue, 26 Sep 2023 21:36:09 +0800
Subject: [PATCH 3/4] Drop usage of Qt 6.5 colorScheme API on Linux

Qt 6.5 added a colorScheme API and telegram desktop has started
to use that instead of the custom portal listener.

However, the current situation w/ that is on Linux is not ideal:
The implementation inside Qt has issues and only really works with
the Qt patches from telegram, and 3rd party platform themes
haven't yet implemented this new API, with the notable exception
being the now-unmaintained QGnomePlatform.

Revert back to the pre-6.5 behavior until the situation improves.
---
 Telegram/SourceFiles/platform/linux/integration_linux.cpp | 4 ----
 Telegram/SourceFiles/platform/linux/specific_linux.cpp    | 2 +-
 Telegram/SourceFiles/platform/platform_specific.h         | 2 +-
 Telegram/SourceFiles/window/themes/window_theme.cpp       | 2 +-
 4 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/Telegram/SourceFiles/platform/linux/integration_linux.cpp b/Telegram/SourceFiles/platform/linux/integration_linux.cpp
index 147a9f03c..9828b829b 100644
--- a/Telegram/SourceFiles/platform/linux/integration_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/integration_linux.cpp
@@ -189,14 +189,11 @@ private:
 
 	const gi::ref_ptr<Application> _application;
 	XdpInhibit::InhibitProxy _inhibitProxy;
-#if QT_VERSION < QT_VERSION_CHECK(6, 5, 0)
 	base::Platform::XDP::SettingWatcher _darkModeWatcher;
-#endif // Qt < 6.5.0
 };
 
 LinuxIntegration::LinuxIntegration()
 : _application(MakeApplication())
-#if QT_VERSION < QT_VERSION_CHECK(6, 5, 0)
 , _darkModeWatcher(
 	"org.freedesktop.appearance",
 	"color-scheme",
@@ -205,7 +202,6 @@ LinuxIntegration::LinuxIntegration()
 			Core::App().settings().setSystemDarkMode(value.get_uint32() == 1);
 		});
 })
-#endif // Qt < 6.5.0
 {
 	LOG(("Icon theme: %1").arg(QIcon::themeName()));
 	LOG(("Fallback icon theme: %1").arg(QIcon::fallbackThemeName()));
diff --git a/Telegram/SourceFiles/platform/linux/specific_linux.cpp b/Telegram/SourceFiles/platform/linux/specific_linux.cpp
index 32437fb77..d67b84184 100644
--- a/Telegram/SourceFiles/platform/linux/specific_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/specific_linux.cpp
@@ -533,7 +533,7 @@ QString SingleInstanceLocalServerName(const QString &hash) {
 #endif // !Q_OS_LINUX || Qt < 6.2.0
 }
 
-#if QT_VERSION < QT_VERSION_CHECK(6, 5, 0)
+#if 1
 std::optional<bool> IsDarkMode() {
 	auto result = base::Platform::XDP::ReadSetting(
 		"org.freedesktop.appearance",
diff --git a/Telegram/SourceFiles/platform/platform_specific.h b/Telegram/SourceFiles/platform/platform_specific.h
index ce839beb1..23c152532 100644
--- a/Telegram/SourceFiles/platform/platform_specific.h
+++ b/Telegram/SourceFiles/platform/platform_specific.h
@@ -51,7 +51,7 @@ void NewVersionLaunched(int oldVersion);
 [[nodiscard]] bool PreventsQuit(Core::QuitReason reason);
 [[nodiscard]] QString ExecutablePathForShortcuts();
 
-#if QT_VERSION < QT_VERSION_CHECK(6, 5, 0)
+#if 1
 [[nodiscard]] std::optional<bool> IsDarkMode();
 #endif // Qt < 6.5.0
 
diff --git a/Telegram/SourceFiles/window/themes/window_theme.cpp b/Telegram/SourceFiles/window/themes/window_theme.cpp
index 88bbe9cf2..e6915b70b 100644
--- a/Telegram/SourceFiles/window/themes/window_theme.cpp
+++ b/Telegram/SourceFiles/window/themes/window_theme.cpp
@@ -526,7 +526,7 @@ void ChatBackground::start() {
 		checkUploadWallPaper();
 	}, _lifetime);
 
-#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
+#if 0
 	rpl::single(
 		QGuiApplication::styleHints()->colorScheme()
 	) | rpl::then(
-- 
2.46.0

